src/ros_license_toolkit/checks.py

```diff
@@ -135,48 +147,65 @@ def _check(self, package: Package):
             if not license_tag.has_license_text_file():
                 license_tags_without_license_text[
                     license_tag] = "No license text file defined."
+                missing_license_texts_status[license_tag] = Status.FAILURE
                 continue
             license_text_file = license_tag.get_license_text_file()
             if not os.path.exists(
                     os.path.join(package.abspath, license_text_file)):
                 license_tags_without_license_text[license_tag] =\
                     f"License text file '{license_text_file}' does not exist."
+                missing_license_texts_status[license_tag] = Status.FAILURE
                 continue
             if license_text_file not in found_license_texts:
                 license_tags_without_license_text[license_tag] =\
                     f"License text file '{license_text_file}' not included" +\
                     " in scan results."
+                missing_license_texts_status[license_tag] = Status.FAILURE
                 continue
             if not is_license_text_file(
                     found_license_texts[license_text_file]):
                 license_tags_without_license_text[license_tag] =\
                     f"License text file '{license_text_file}' is not " +\
                     "recognized as license text."
+                missing_license_texts_status[license_tag] = Status.FAILURE
                 continue
             actual_license: Optional[str] = get_spdx_license_name(
                 found_license_texts[license_text_file])
             if actual_license is None:
                 license_tags_without_license_text[license_tag] =\
                     f"License text file '{license_text_file}'" +\
                     " is not recognized as license text."
+                missing_license_texts_status[license_tag] = Status.FAILURE
                 continue
             if actual_license != license_tag.get_license_id():
                 license_tags_without_license_text[license_tag] =\
                     f"License text file '{license_text_file}' is " +\
                     f"of license {actual_license} but should be " +\
                     f"{license_tag.get_license_id()}."
+                missing_license_texts_status[license_tag] = Status.WARNING
                 continue
         if len(license_tags_without_license_text) > 0:
-            self._failed(
-                "The following license tags do not have a valid license text "
-                "file:\n" + "\n".join(
-                    [f"  '{x[0]}': {x[1]}" for x in
-                        license_tags_without_license_text.items()]))
+            if all(status == Status.WARNING
+                   for status in missing_license_texts_status.values()):
+                self._warning(
+                    "The following license tags do not "
+                    "have a valid license text "
+                    "file:\n" + "\n".join(
```

Comment:
```suggestion
                    "Since they are not in the SPDX list, "
                    "we can not check of these tags have the correct "
                    "licens text:\n" + "\n".join(
```
or something like this